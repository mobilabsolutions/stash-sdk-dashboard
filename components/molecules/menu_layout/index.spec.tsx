import { testRender, deepRender } from '../../../test_utils'
import MenuLayout from './index'
import { useState } from 'react'

const TestMenu = () => {
  const [active, setActive] = useState(0)
  return (
    <MenuLayout
      active={active}
      setActive={setActive}
      menu={[
        {
          title: 'String Title',
          render: () => <span>Content 1</span>
        },
        {
          title: () => <h1>Function Title</h1>,
          render: () => <span>Content 2</span>
        },
        {
          title: () => <h1>Function Title 1</h1>,
          render: () => <span>Content 3</span>
        }
      ]}
    />
  )
}

describe('Menu layout', () => {
  test('should render', () => {
    testRender(TestMenu, {})
  })
  test('should change content', () => {
    const { getByTestId, baseElement } = deepRender(TestMenu, {})
    getByTestId('menu-2').click()
    expect(baseElement.firstChild).toMatchSnapshot()
  })
})
