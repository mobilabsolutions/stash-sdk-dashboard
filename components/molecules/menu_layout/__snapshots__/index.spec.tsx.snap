// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu layout should change content 1`] = `
<div>
  <div
    class="menu_layout__Wrapper-sc-110xcxw-0 bbHmeo"
  >
    <div
      class="menu-area"
    >
      <div
        class="menu-item"
        data-testid="menu-0"
      >
        String Title
      </div>
      <div
        class="menu-item"
        data-testid="menu-1"
      >
        <h1>
          Function Title
        </h1>
      </div>
      <div
        class="menu-item"
        data-testid="menu-2"
      >
        <h1>
          Function Title 1
        </h1>
      </div>
    </div>
    <div
      class="vertical_scroll_container__Container-bnn76i-0 iKbzRd"
    >
      <div
        class="vertical_scroll_container__MarginContainer-bnn76i-1 eYFwBI"
      >
        <span>
          Content 3
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Menu layout should render 1`] = `
<div
  className="menu_layout__Wrapper-sc-110xcxw-0 bbHmeo"
>
  <div
    className="menu-area"
  >
    <div
      className="menu-item"
      data-testid="menu-0"
      onClick={[Function]}
    >
      String Title
    </div>
    <div
      className="menu-item"
      data-testid="menu-1"
      onClick={[Function]}
    >
      <h1>
        Function Title
      </h1>
    </div>
    <div
      className="menu-item"
      data-testid="menu-2"
      onClick={[Function]}
    >
      <h1>
        Function Title 1
      </h1>
    </div>
  </div>
  <div
    className="vertical_scroll_container__Container-bnn76i-0 iKbzRd"
  >
    <div
      className="vertical_scroll_container__MarginContainer-bnn76i-1 eYFwBI"
    >
      <span>
        Content 1
      </span>
    </div>
  </div>
</div>
`;
